### PR TITLE
feat: add rewarded interstitial ad

### PR DESCRIPTION
**Pull request by wjaykim**
### Description

This PR adds support for [Rewarded Interstitial Ad](https://developers.google.com/admob/android/rewarded-interstitial).

### Release Summary

Added rewarded interstitial ad

### Checklist

- I read the [Contributor Guide](https://github.com/invertase/react-native-google-mobile-ads/blob/main/CONTRIBUTING.md)
  and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [x] `e2e` tests added or updated in `__tests__e2e__`
  - [ ] `jest` tests added or updated in `__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-google-mobile-ads` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: src/ads/MobileAd.ts [hunk 1]
```diff
@@ -158,15 +158,26 @@ export abstract class MobileAd implements MobileAdInterface {
     return this.constructor.name;
   }
 
+  protected get _camelCaseType() {
```
Comment (mikehardy) [inline src/ads/MobileAd.ts:161]: nice

File: src/types/GoogleMobileAdsNativeModule.ts [hunk 2]
```diff
@@ -3,14 +3,23 @@ import { RequestConfiguration } from './RequestConfiguration';
 import { RequestOptions } from './RequestOptions';
 import { AdShowOptions } from './AdShowOptions';
 
+type AdLoadFunction = (requestId: number, adUnitId: string, requestOptions: RequestOptions) => void;
+type AdShowFunction = (
```
Comment (mikehardy) [inline src/types/GoogleMobileAdsNativeModule.ts:7]: nice!

**Comment by mikehardy** [comment]
I just sent an invite to be a maintainer in the repo @wjaykim - you've certainly earned it! No pressure and I understand if you don't want to, but one of my goals (as you know :-) ) is to have one repo with multiple maintainers so that we have redundancy and support among each other. So I'd love it if you accept even if it's just to serve as a backup backup for Dylan and I

If you do accept then once this PR goes in you may enjoy the slightly-nervous but mostly-fun feeling of running the publish action on main as well, to launch it :rocket: :laughing:

**Comment by mikehardy** [comment]
Ah great, then this is clean enough to eat from :laughing: - fantastic re: accepting maintainer invite too. We also have a channel in the Invertase slack space if you want an invite there. Again, totally at your option, but if Dylan and I ever need to chat real time we use that, and it is a way to communicate with the other Invertase folks as needed (for more restricted github permissions or also just to have other people to ask questions from whenever they come up). Let me know if that's interesting and if so we can send an invite to your preferred address.

As for this PR, I'll merge it, and if you watch the Actions tab - whenever they are all green, select the "Publish" one, do a manual run on the main branch and....apart from hoping it works, that will launch it following semantic release commit log examination + auto-versioning based on detected commits. Then you will also have the power here and know how to do releases as needed

**Comment by wjaykim** [comment]
@mikehardy You can invite me to slack, the mail address is wj118797@gmail.com

**Pull request merged**